connector/interface.go

```diff
@@ -60,6 +60,11 @@ type ConnectorConfig interface {
 	Connector(ns url.URL, loginFunc oidc.LoginFunc, tpls *template.Template) (Connector, error)
 }
 
+// Groups is an optional interface implemented by a Connector.
+type Groups interface {
+	Groups(fullUserID string) ([]string, error)
```

Comment:
Like being able to differentiated between connectors that support this and a call returning an error. Would prefer to either embed the Connector interface or adjust the naming to make it more explicit. 

Would like to use this pattern later for connectors that accept passwords (LDAP and local). So another interface like this

```
type PasswordConnector interface {
    Login(username, password) (identity *oidc.Identity, correctCreds bool, err error)
}
```

Then detect connectors that accept passwords so the connectors themselves stop rendering HTML.